app/uk/gov/hmrc/accessibilitystatementfrontend/controllers/StatementController.scala

```diff
@@ -55,11 +57,21 @@ class StatementController @Inject() (
         language = messagesApi.preferred(request).lang
       ) match {
         case Some((accessibilityStatement, language)) =>
+          val safeReferrerUrl: Option[SafeRedirectUrl] = referrerUrl match {
```

Comment:
I could be missing something but is there a reason for case matching over mapping on this `Option`?

Reply:
It's a bit awkward that in the `Left()` case all we get is an error message, so we then have to turn the original `RedirectUrl` into a `SafeRedirectUrl` via `.get(UnsafePermitAll)`, and I couldn't see a way to avoid having to then `.get` the resulting `Option` - so went for a pattern match.  But maybe I missed a trick?

Reply:
nvm - have refactored now... put it down to my Monday brain 🧠